lib/ua_update_messaging.py

```diff
@@ -192,49 +220,64 @@ def write_apt_and_motd_templates(cfg: config.UAConfig, series: str) -> None:
     motd_infra_no_pkg_file = ExternalMessage.MOTD_INFRA_NO_PKGS.value
     motd_infra_pkg_file = ExternalMessage.MOTD_INFRA_PKGS.value
     no_warranty_file = ExternalMessage.UBUNTU_NO_WARRANTY.value
+    msg_dir = os.path.join(cfg.data_dir, "messages")
 
     apps_cls = entitlements.ENTITLEMENT_CLASS_BY_NAME["esm-apps"]
     apps_inst = apps_cls(cfg)
     config_allow_beta = util.is_config_value_true(
         config=cfg.cfg, path_to_value="features.allow_beta"
     )
-    apps_not_beta = bool(config_allow_beta or not apps_cls.is_beta)
+    apps_valid = bool(config_allow_beta or not apps_cls.is_beta)
     infra_cls = entitlements.ENTITLEMENT_CLASS_BY_NAME["esm-infra"]
     infra_inst = infra_cls(cfg)
 
     expiry_status, remaining_days = get_contract_expiry_status(cfg)
 
-    if series != "trusty" and apps_not_beta:
-        _write_esm_service_msg_templates(
-            cfg,
-            apps_inst,
-            expiry_status,
-            remaining_days,
-            apps_pkg_file,
-            apps_no_pkg_file,
-            motd_apps_pkg_file,
-            motd_apps_no_pkg_file,
-            no_warranty_file,
+    enabled_status = ApplicationStatus.ENABLED
+    msg_esm_apps = False
+    msg_esm_infra = False
+    if util.is_active_esm(series):
+        if expiry_status in (
+            ContractExpiryStatus.EXPIRED,
+            ContractExpiryStatus.NONE,
+        ):
+            no_warranty_msg = MESSAGE_UBUNTU_NO_WARRANTY
```

Comment:
My initial thought was that we would print `no_warranty` on "active esm releases" when either (a) esm-infra is not enabled, or (b) esm-infra is enabled but expired. Is this check equivalent to that? Specifically, is the expiry_status always NONE when its not enabled?

Or is my expectation not in line with what we actually want here?

Reply:
> My initial thought was that we would print `no_warranty` on "active esm releases" when either (a) esm-infra is not enabled, or (b) esm-infra is enabled but expired. Is this check equivalent to that? Specifically, is the expiry_status always NONE when its not enabled?
> 
> Or is my expectation not in line with what we actually want here?

Your expectation is correct:
a: need to report "no warranty" when esm-infra not enabled on an is_active_esm release (incomplete handled by caught by ContractExpiryStatus.NONE (meaning not attached).  )
b. esm-infra enabled by expired, (handled by  everything being expired per ContractExpiryStatus.EXPIRED)

So, yes, it seems I need to fix the case in "a." about esm-infra not being enabled too and emit that 'no warranty' good catch
b.

Reply:
Pushed fix, we will now also emit no warranty if contract is still active, but esm-infra is not enabled on active ESM releases.